diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3fe70f98..3da46683 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,10 +23,20 @@ add_library(binaryninjaapi STATIC ${BN_API_SOURCES})
 target_include_directories(binaryninjaapi
     PUBLIC ${PROJECT_SOURCE_DIR})
 
-find_package(BinaryNinjaCore REQUIRED)
-target_link_libraries(binaryninjaapi PUBLIC ${BinaryNinjaCore_LIBRARIES})
-target_link_directories(binaryninjaapi PUBLIC ${BinaryNinjaCore_LIBRARY_DIRS})
-target_compile_definitions(binaryninjaapi PUBLIC ${BinaryNinjaCore_DEFINITIONS})
+find_package(BinaryNinjaCore)
+if(BinaryNinjaCore_FOUND)
+    target_link_libraries(binaryninjaapi PUBLIC ${BinaryNinjaCore_LIBRARIES})
+    target_link_directories(binaryninjaapi PUBLIC ${BinaryNinjaCore_LIBRARY_DIRS})
+    target_compile_definitions(binaryninjaapi PUBLIC ${BinaryNinjaCore_DEFINITIONS})
+else()
+    if(APPLE)
+        target_link_options(binaryninjaapi PUBLIC -undefined dynamic_lookup)
+    elseif(MSVC)
+        target_link_options(binaryninjaapi PUBLIC "LINKER:/FORCE:UNRESOLVED")
+    else()
+        target_link_options(binaryninjaapi PUBLIC "LINKER:--allow-shlib-undefined")
+    endif()
+endif()
 
 set_target_properties(binaryninjaapi PROPERTIES
     CXX_STANDARD 17
@@ -71,16 +81,18 @@ function(bn_install_plugin target)
         list(APPEND CMAKE_MODULE_PATH "${BN_API_SOURCE_DIR}/cmake")
 
         # BinaryNinjaCore has the user plugins dir define that we want
-        find_package(BinaryNinjaCore REQUIRED)
-        if(WIN32)
-            install(TARGETS ${target} RUNTIME
-                DESTINATION ${BinaryNinjaCore_USER_PLUGINS_DIR})
-
-            install(FILES $<TARGET_PDB_FILE:${target}>
-                DESTINATION ${BinaryNinjaCore_USER_PLUGINS_DIR} OPTIONAL)
-        else()
-            install(TARGETS ${target} LIBRARY
-                DESTINATION ${BinaryNinjaCore_USER_PLUGINS_DIR})
+        find_package(BinaryNinjaCore)
+        if(BinaryNinjaCore_FOUND)
+            if(WIN32)
+                install(TARGETS ${target} RUNTIME
+                    DESTINATION ${BinaryNinjaCore_USER_PLUGINS_DIR})
+
+                install(FILES $<TARGET_PDB_FILE:${target}>
+                    DESTINATION ${BinaryNinjaCore_USER_PLUGINS_DIR} OPTIONAL)
+            else()
+                install(TARGETS ${target} LIBRARY
+                    DESTINATION ${BinaryNinjaCore_USER_PLUGINS_DIR})
+            endif()
         endif()
     endif()
 endfunction()
diff --git a/ui/linearview.h b/ui/linearview.h
index fd19cc59..7c988edd 100644
--- a/ui/linearview.h
+++ b/ui/linearview.h
@@ -12,7 +12,7 @@
 #include "uicontext.h"
 #include "instructionedit.h"
 #include "ilchooser.h"
-#include <assembledialog.h>
+#include "binaryninja-api/ui/assembledialog.h"
 
 #define LINEAR_VIEW_UPDATE_CHECK_INTERVAL 200
 #define MAX_STRING_TYPE_LENGTH            1048576
@@ -220,7 +220,7 @@ class BINARYNINJAUIAPI LinearView : public QAbstractScrollArea, public View, pub
 
 	QTimer* m_hoverTimer;
 	QPointF m_previewPos;
-
+public:
 	ContextMenuManager* m_contextMenuManager;
 	QPointer<CommentDialog> m_commentDialog;
 
diff --git a/ui/menus.h b/ui/menus.h
index 894a1f79..231369de 100644
--- a/ui/menus.h
+++ b/ui/menus.h
@@ -25,6 +25,7 @@ class BINARYNINJAUIAPI ContextMenuManager : public QObject
 {
 	Q_OBJECT
 
+public:
 	QWidget* m_parent;
 	QMenu* m_menu;
 	MenuInstance* m_instance;
diff --git a/ui/metadatachoicedialog.h b/ui/metadatachoicedialog.h
index aac8da3d..0ebb3f12 100644
--- a/ui/metadatachoicedialog.h
+++ b/ui/metadatachoicedialog.h
@@ -225,7 +225,7 @@ public:
 class BINARYNINJAUIAPI MetadataChoiceDialog : public QDialog, public ManagedTableDelegate, public EntryItemMetadataViewDelegate {
 	Q_OBJECT
 
-protected:
+public:
 
 	std::vector<EntryItem> m_entries;
 
diff --git a/ui/sidebar.h b/ui/sidebar.h
index ead1ca8d..a75d11d0 100644
--- a/ui/sidebar.h
+++ b/ui/sidebar.h
@@ -38,7 +38,7 @@ class BINARYNINJAUIAPI SidebarWidget : public QWidget
 {
 	Q_OBJECT
 
-  protected:
+  public:
 	QString m_title;
 	UIActionHandler m_actionHandler;
 	ContextMenuManager* m_contextMenuManager = nullptr;
