diff --git a/ui/linearview.h b/ui/linearview.h
index fd19cc59..e9f9eb9a 100644
--- a/ui/linearview.h
+++ b/ui/linearview.h
@@ -220,7 +220,7 @@ class BINARYNINJAUIAPI LinearView : public QAbstractScrollArea, public View, pub
 
 	QTimer* m_hoverTimer;
 	QPointF m_previewPos;
-
+public:
 	ContextMenuManager* m_contextMenuManager;
 	QPointer<CommentDialog> m_commentDialog;
 
diff --git a/ui/menus.h b/ui/menus.h
index 894a1f79..231369de 100644
--- a/ui/menus.h
+++ b/ui/menus.h
@@ -25,6 +25,7 @@ class BINARYNINJAUIAPI ContextMenuManager : public QObject
 {
 	Q_OBJECT
 
+public:
 	QWidget* m_parent;
 	QMenu* m_menu;
 	MenuInstance* m_instance;
diff --git a/ui/metadatachoicedialog.h b/ui/metadatachoicedialog.h
index aac8da3d..0ebb3f12 100644
--- a/ui/metadatachoicedialog.h
+++ b/ui/metadatachoicedialog.h
@@ -225,7 +225,7 @@ public:
 class BINARYNINJAUIAPI MetadataChoiceDialog : public QDialog, public ManagedTableDelegate, public EntryItemMetadataViewDelegate {
 	Q_OBJECT
 
-protected:
+public:
 
 	std::vector<EntryItem> m_entries;
 
diff --git a/ui/sidebar.h b/ui/sidebar.h
index ead1ca8d..a75d11d0 100644
--- a/ui/sidebar.h
+++ b/ui/sidebar.h
@@ -38,7 +38,7 @@ class BINARYNINJAUIAPI SidebarWidget : public QWidget
 {
 	Q_OBJECT
 
-  protected:
+  public:
 	QString m_title;
 	UIActionHandler m_actionHandler;
 	ContextMenuManager* m_contextMenuManager = nullptr;
